units/tasks_restart.go

```diff
@@ -0,0 +1,52 @@
+package units
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/logging"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+type restartTasksJob struct {
+	job.Base  `bson:"job_base" json:"job_base" yaml:"job_base"`
+	logger    grip.Journaler
+	startTime time.Time
+	endTime   time.Time
+	user      string
+	opts      model.RestartTaskOptions
+}
+
+// NewTasksRestartJob creates a job to restart failed tasks in a time range
```

Comment:
- register the job. 
- populate the job.Base: JobType field, etc.) 
- make the job serializable (export fields, add tags).

Reply:
done